torch/_inductor/ir.py

```diff
@@ -604,9 +607,9 @@ def _is_static(x):
                 "var_unnormalized",
             }
             and config.split_reductions
-            and all(_is_static(r) for r in ranges)
-            and all(_is_static(r) for r in reduction_ranges)
-            and _is_static(reduction_numel)
+            # We don't support unbacked symints
+            and _is_static(reduction_numel_hint)
+            and _is_static(numel_hint)
```

Comment:
Can it be the case that `numel` is static but `reduction_numel` is not?

Reply:
In fact, we probably just want `_is_static(reduction_numel_hint)` for the general case. Then, if we also have `numel_hint` static even better, but it's not 100% necessary for the main optimisation, I don't think.

Reply:
> Can it be the case that `numel` is static but `reduction_numel` is not?

`numel` here is actually the number of output elements, so doesn't include the reduced dimensions.

> n fact, we probably just want `_is_static(reduction_numel_hint)` for the general case. Then, if we also have `numel_hint` static even better, but it's not 100% necessary for the main optimisation, I don't think.

`numel_hint` is used in conditionals like `numel_hint >= num_sm * 2 * 32` when deciding on the numbers of splits, so we need to have a concrete value.

Reply:
btw as a further step we could use `bound_sympy` to deal with unbacked SymInts but thats more than I need at this point to get cumsum working.

Reply:
`bound_sympy` does not currently deal with unbacked SymInts, but may be able to do something in some cases when https://github.com/pytorch/pytorch/pull/106568 is merged.